jwst/ramp_fitting/tests/test_ramp_fit_cases.py

```diff
@@ -0,0 +1,541 @@
+    o_true = [1., 56.870003, 0.01818182, 1.0691562, 15., 56.870003, 5.,
+              0.84580624 ]
+
+    new_mod, int_mod, opt_mod, gls_opt_mod = ramp_fit( RampMod, 1024*300., True,
+                                                RnMod, GMod, 'OLS', 'optimal' )
+
+    assert_pri( p_true, new_mod )
+    assert_opt( o_true, opt_mod )
+
+
+def assert_pri( p_true, new_mod ):
+    """
+    Compare true and fit values of primary output for extensions
+    SCI, DQ, ERR, VAR_POISSSON, VAR_RNOISE.
+    """
+    
+    npt.assert_allclose( new_mod.data,        p_true[0], atol=1E-6 )
+    npt.assert_allclose( new_mod.dq,          p_true[1], atol=1E-1 )
+    npt.assert_allclose( new_mod.err,         p_true[2], atol=1E-6 )
+    npt.assert_allclose( new_mod.var_poisson, p_true[3], atol=1E-6 )
+    npt.assert_allclose( new_mod.var_rnoise,  p_true[4], atol=1E-6 )
+
+    return None
+
+
+def assert_opt( o_true, opt_mod ):
+    """
+    Compare true and fit values of optional output for extensions SLOPE,
+    SIGSLOPE, VAR_POISSSON, VAR_RNOISE, YINT, SIGYINT, PEDESTAL, and WEIGHTS.
+    Selecting the particular (and only) ramp in the optional output, which is
+    [0,:,0,0]
+    """
+
+    opt_slope = opt_mod.slope[0,:,0,0]
+    opt_sigslope = opt_mod.sigslope[0,:,0,0]
+    opt_var_poisson = opt_mod.var_poisson[0,:,0,0]
+    opt_var_rnoise = opt_mod.var_rnoise[0,:,0,0]
+    opt_yint = opt_mod.yint[0,:,0,0]
+    opt_sigyint = opt_mod.sigyint[0,:,0,0]
+    opt_pedestal = opt_mod.pedestal[:,0,0]
+    opt_weights = opt_mod.weights[0,:,0,0]
+
+    npt.assert_allclose( opt_slope, o_true[0], atol=1E-6 )
+    npt.assert_allclose( opt_sigslope, o_true[1], atol=1E-6 )
+    npt.assert_allclose( opt_var_poisson, o_true[2], atol=1E-6 )
+    npt.assert_allclose( opt_var_rnoise, o_true[3], atol=1E-6 )
+    npt.assert_allclose( opt_yint, o_true[4], atol=1E-6 )
+    npt.assert_allclose( opt_sigyint, o_true[5], atol=1E-6 )
+    npt.assert_allclose( opt_pedestal, o_true[6], atol=1E-6 )
+    npt.assert_allclose( opt_weights, o_true[7], atol=1E-6 )
+
+    return None
+
+
+def set_scalars():
+    """
+    Set needed scalars for the size of the dataset, and other values needed for
+    the fit.
+    """
+
+    ngroups = 10
+    nints = 1
+    nrows = 1
+    ncols = 1
+    deltatime = 10.0
+    gain = 5.5
+    readnoise = 10.34
+
+    return ngroups, nints, nrows, ncols, deltatime, gain, readnoise
+
+
+def create_mod_arrays(ngroups, nints, nrows, ncols, deltatime, gain, readnoise):
+    """
+    For an input datacube (arbitrarily chosen to be MIRI), create arrays having
+    the specified dimensions for the pixel DQ, the group DQ, and the
+    ERR extensions, and create datamodels for the ramp, readnoise, and gain.
+    """ 
+
+    gain = np.ones(shape=(nrows, ncols), dtype=np.float64) * gain
+    err = np.zeros(shape=(nints, ngroups, nrows, ncols), dtype=np.float64)
+    data = np.zeros(shape=(nints, ngroups, nrows, ncols), dtype=np.float64)
```

Comment:
When we use the ramp_fit routines in the pipeline step we aren't actually passing it float64 data, are we? I thought the data model .data and .err arrays were only float32. Will that bias the test results at all, by not testing in exactly the same way things are used in the actual step?